src/IK/src/JointLimitsTask.cpp

```diff
@@ -0,0 +1,266 @@
+                     errorPrefix,
+                     m_kinDyn->getNrOfDegreesOfFreedom() + 6,
+                     robotVelocityVariable.size);
+        return false;
+    }
+
+
+    // resize the matrices
+    m_A.resize(2 * m_kinDyn->getNrOfDegreesOfFreedom(), variablesHandler.getNumberOfVariables());
+    m_A.setZero();
+    m_b.resize(2 * m_kinDyn->getNrOfDegreesOfFreedom());
+    m_b.head(m_kinDyn->getNrOfDegreesOfFreedom()).setConstant(OsqpEigen::INFTY);
+    m_b.tail(m_kinDyn->getNrOfDegreesOfFreedom()).setConstant(-OsqpEigen::INFTY);
+
+    // the submatrix associated to the robot velocity
+    //      _                _
+    // A = | 0_{6x6} +I_{nxn} |
+    //     | 0_{6x6} -I_{nxn} |
+    //     |_                _|
+    iDynTree::toEigen(this->subA(robotVelocityVariable))
+        .topRightCorner(m_kinDyn->getNrOfDegreesOfFreedom(), m_kinDyn->getNrOfDegreesOfFreedom()).diagonal().setConstant(1);
+    iDynTree::toEigen(this->subA(robotVelocityVariable))
+        .bottomRightCorner(m_kinDyn->getNrOfDegreesOfFreedom(), m_kinDyn->getNrOfDegreesOfFreedom()).diagonal().setConstant(-1);
+
+    return true;
+}
+
+bool JointLimitsTask::initialize(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> paramHandler)
+{
+    constexpr auto errorPrefix = "[JointLimitsTask::initialize]";
+
+    if (m_kinDyn == nullptr || !m_kinDyn->isValid())
+    {
+        log()->error("{} KinDynComputations object is not valid.", errorPrefix);
+        return false;
+    }
+
+    auto ptr = paramHandler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} The parameter handler is not valid.", errorPrefix);
+        return false;
+    }
+
+    if (!ptr->getParameter("robot_velocity_variable_name", m_robotVelocityVariableName))
+    {
+        log()->error("{} Error while retrieving the robot velocity variable.", errorPrefix);
+        return false;
+    }
+
+    // set the gains for the controllers
+    m_klim.resize(m_kinDyn->getNrOfDegreesOfFreedom());
+    if (!ptr->getParameter("klim", m_klim))
+    {
+        log()->error("{} Error while retrieving the proportional gain.", errorPrefix);
+        return false;
+    }
+
+    if (m_klim.minCoeff() < 0 || m_klim.maxCoeff() > 1)
+    {
+        log()->error("{} The proportial gains must contain only positive numbers lower than 1.",
+                     errorPrefix);
+        return false;
+    }
+
+    if (!ptr->getParameter("sampling_time", m_samplingTime) || m_samplingTime < 0)
+    {
+        log()->error("{} Error while retrieving the sampling time. Please remember that must be a "
+                     "positive number",
+                     errorPrefix);
+        return false;
+    }
+
+    // check if the
+    bool useModelLimits = true;
+    if (!ptr->getParameter("use_model_limits", useModelLimits))
+    {
+        log()->error("{} Error while retrieving the 'use_model_limits' parameter", errorPrefix);
+        return false;
+    }
```

Comment:
If there is an error with a specific parameter, I would specify in the message the exact name of the parameter that triggered the error

Reply:
Done 2865e2b6b4fc47582008579d5bee78e4e2877e39